### DISCUSSION TITLE
Error: Command not found: ./vercel-dev-server-go

### DISCUSSION THREAD

**Discussion by haasal** [vercel/community#1265] [Help] [answered]
### Summary

I installed the vercel cli with

```
pnpm install vercel
```

into an astro project.
Then I added the default `index.go` file into the api folder in my project directory.
When starting the dev server with `vercel dev` and try to access `localhost:3000/api/` I get a 502 response and vercel cli outputs:

```
Error: Command not found: ./vercel-dev-server-go
Please ensure that `./vercel-dev-server-go` is properly installed
```

I have no Idea what could be causing this. Go is installed and in the PATH.
I'm running Linux 6.1.5-arch2-1

**Comment by rafbgarcia** [comment]
I just ran into the same issue.

```
➜  govercel vercel dev --listen  8000
Vercel CLI 28.5.6
> Ready! Available at http://localhost:8000
Error: Command not found: ./vercel-dev-server-go
Please ensure that `./vercel-dev-server-go` is properly installed
Learn More: https://vercel.link/command-not-found
```

**Comment by amyegan** [comment]
[Vercel CLI](https://vercel.com/docs/cli#) should be installed globally. Do you still get the error after running `pnpm i -g vercel`?

**Reply by haasal** [reply]
Yes. The issue persists

**Reply by haasal** [reply]
Currently I have go 1.19.5 installed. Do I have to downgrade to some specific version?

**Reply by amyegan** [reply]
The Go Runtime automatically detects the `go.mod` file at the root of your Project to determine the Go version and install dependencies. If `go.mod` is missing or the version is not defined, the default version `1.16` is used. Version `1.19.5` should be fine.

Since you already have both Go and Vercel CLI installed, I'm not sure what else is causing this error. I can't find much about it, so I'm reaching out to the engineering team for more info and I'll let you know when I learn more

**Reply by smaeda-ks** [reply]
@haasal Could you share your debug output logs as well?

`vercel dev --debug`

and which Vercel CLI version you're using? `vercel -v`

**Reply by smaeda-ks** [reply]
https://github.com/vercel/vercel/pull/9320

^^ This change will be shipped in the next CLI version (it's not a bug fix, but for better error message).
As you can see note in the PR description, we suspect that your go binary is not found in the `PATH`.

Could you double-check that?

**Reply by haasal** [reply]
Go is in PATH

**Comment by leandrofinger** [comment, marked as answer]
I had the same issue, so I deployed the application to check if the issue was related only to CLI. The build process complained about the `go` version specified in `go.mod` since `1.19` is not supported by `tiny` lib. Once I changed `go.mod` version to `1.18` both build process and `vercel dev` started working correctly.

PS: I didn't downgraded my go installed version that is `1.19.5`, I just changed its version in `go.mod` file

**Reply by amyegan** [reply]
Thanks for sharing this, Leandro! 

@haasal @rafbgarcia Does it the error go away if you switch to `1.18` in `go.mod`?

**Reply by rafbgarcia** [reply]
Actually, now it's working fine for me, even on `1.19`.

I apologize, I think I missed something in my previous test.

<img width="1439" alt="Cursor_and_vercel_dev_--listen_8000_and_index_go_—_gonext_and_localhost_8000_api" src="https://user-images.githubusercontent.com/1904314/214585228-4a0569e8-7c65-4481-b47b-44d30339e07f.png">

**Reply by rafbgarcia** [reply]
Thank you for the help, @amyegan!

**Reply by haasal** [reply]
Thank you! It now works. I think the issue was that the go.mod file has to be in the root of the project. Without a go.mod file it didn't work at all. This wasn't mentioned at all in the docs.
